j-lawyer-client/src/de/costache/calendar/util/CalendarUtil.java

```diff
@@ -218,13 +218,43 @@ public static Date roundDateToHalfAnHour(Date date, boolean roundUp) {
 
         if (time <= 30) {
             c.set(Calendar.MINUTE, roundUp ? 30 : 0);
-        } else if (time > 30) {
+        } else {
             c.set(Calendar.MINUTE, roundUp ? 0 : 30);
-            if(roundUp)
+            if (roundUp)
+                c.add(Calendar.HOUR, 1);
+        }
+        c.set(Calendar.SECOND, 0);
+        c.set(Calendar.MILLISECOND, 0);
+        return c.getTime();
+    }
+
+    public static Date roundDateToHalfAnHour2(Date date, boolean roundUp) {

```

Comment:
is this method used somewhere?

Reply:
if it is supposed to remain, having  a (very brief) comment on the difference in behaviour would be beneficial

Reply:
Removed, sorry for the confusion, this was accidentally added. 
I first searched for an issue in the rounding method, but then solved it by storing the selected date directly.
Previously it was Coordinates => Rounded Date => Coordinates on Move and then on Release it was Coordinates => Rounded Date again leading to different results.